modules/src/ics04_channel/handler/chan_open_init.rs

```diff
@@ -0,0 +1,196 @@
+    }
+
+    // An IBC connection running on the local (host) chain should exist.
+    if ctx
+        .connection_state(&msg.channel().connection_hops()[0])
+        .is_none()
+    {
+        return Err(Kind::MissingConnection(msg.channel().connection_hops()[0].clone()).into());
+    }
+
+    // TODO: Check that `version` is non empty but not necessary coherent
+    if msg.channel().version().is_empty() {
+        return Err(Kind::InvalidVersion.into());
+    }
+
+    let new_channel_end = ChannelEnd::new(
+        State::Init,
+        *msg.channel().ordering(),
+        msg.channel().counterparty().clone(),
+        msg.channel().connection_hops().clone(),
+        ctx.get_compatible_versions()[0].clone(),
+    );
+
+    output.log("success: no channel found");
+
+    let result = ChannelResult {
+        port_id: msg.port_id().clone(),
+        channel_id: None,
+        channel_end: new_channel_end,
+        channel_cap: cap_key,
+    };
+
+    output.emit(ChanOpenInit(result.clone()));
+
+    Ok(output.with_result(result))
+}
+
+#[cfg(test)]
+mod tests {
+    use std::convert::TryFrom;
+    use std::str::FromStr;
+
+    use crate::handler::EventType;
+
+    use crate::ics03_connection::msgs::conn_open_init::test_util::get_dummy_msg_conn_open_init;
+
+    use crate::ics03_connection::connection::State as ConnectionState;
+    use crate::ics03_connection::msgs::conn_open_init::MsgConnectionOpenInit;
+    use crate::ics04_channel::channel::State;
+
+    use crate::ics03_connection::connection::ConnectionEnd;
+    use crate::ics03_connection::version::get_compatible_versions;
+    use crate::ics04_channel::handler::{dispatch, ChannelResult};
+    use crate::ics04_channel::msgs::chan_open_init::test_util::get_dummy_raw_msg_chan_open_init;
+
+    use crate::ics04_channel::msgs::chan_open_init::MsgChannelOpenInit;
+    use crate::ics04_channel::msgs::ChannelMsg;
+
+    use crate::ics24_host::identifier::ConnectionId;
+    use crate::mock::context::MockContext;
+
+    #[test]
+    fn chan_open_init_msg_processing() {
+        struct Test {
+            name: String,
+            ctx: MockContext,
+            msg: ChannelMsg,
+            want_pass: bool,
+        }
+
+        let msg_chan_init =
+            MsgChannelOpenInit::try_from(get_dummy_raw_msg_chan_open_init()).unwrap();
+
+        let context = MockContext::default();
+
+        let msg_conn_init =
+            MsgConnectionOpenInit::try_from(get_dummy_msg_conn_open_init()).unwrap();
+
+        let init_conn_end = ConnectionEnd::new(
+            ConnectionState::Init,
+            msg_conn_init.client_id().clone(),
+            msg_conn_init.counterparty().clone(),
+            get_compatible_versions(),
+            msg_conn_init.delay_period,
+        );
+
+        let ccid = <ConnectionId as FromStr>::from_str("defaultConnection-0");
+        let cid = match ccid {
+            Ok(v) => v,
+            Err(_e) => ConnectionId::default(),
+        };
```

Comment:
This might be easier, and maybe it achieves the same thing.

```suggestion
        let ccid = ConnectionId::default();
```

Reply:
The default is not yet versioned. Connection default returns just defaultConnection. I think I should modify it but I didn't want to mess with stuff I didn't do so soon.

Reply:
That makes sense, thanks for the explanation!

Is the `default` ConnectionId going to make the tests fail (due to the missing version number)? If the missing version is not problematic, I'd suggest we still go with `let ccid = ConnectionId::default();`; otherwise, if the missing version poses problems, we can just keep the code here as you wrote it initially.

Reply:
ConnectionId::default() needs to stay "defaultConnection" cause next_connection_id adds the counter as a suffix.  
For example the test in ics26 fails if ConnectionId::default() is versioned, cause the keeper will add the version  to the default connection and for channel a connection needs to be present. 
The function  ics04_channel::channel::test_util::get_dummy_raw_channel_end which returns a channel end with a versioned connection "defaultConnection-0". Here in this test I kind of do what get_dummy_raw_channel_end does in a way that compiles locally.

Reply:
Ok, I understand. This has to do with the [recent changes](#449) on how `ConnectionId::default()` is used. So let's ignore my comment then, and we can refactor this later.